README.md

```diff
@@ -44,6 +46,18 @@ Just use your package managers as you usually would. Run `yarn install` in Yarn
 
 - **If the local project isn't configured for any package manager**, Corepack will assume that you know what you're doing, and will use whatever package manager version has been pinned as "known good release". Check the relevant section for more details.
 
+### When Authoring Packages
+
+Set your package's manager with the `packageManager` field in `package.json`:
+
+```json
+{
+  "packageManager": "yarn@3.2.3+sha224.953c8233f7a92884eee2de69a1b92d1f2ec1655e66d08071ba9a02fa"
+}
+```
+
+Here, `yarn` executable is specified at version `3.2.3`, along with the `sha224` hash of this version for validation. `executable@x.y.z` is required. The sha is optional but strongly recommended as a security practice. Permitted values for the executable are `yarn`, `npm`, and `pnpm`.
```

Comment:
IMO we should not use `executable` here (because e.g. npm ships with several executables, but we only accept `npm` as a value), but I don't feel strongly about this.
```suggestion
Here, `yarn` is the name of the package manager, specified at version `3.2.3`, along with the SHA-224 hash of this version for validation. `packageManager@x.y.z` is required. The hash is optional but strongly recommended as a security practice. Permitted values for the package manager are `yarn`, `npm`, and `pnpm`.
```